letsencrypt/cli.py

```diff
@@ -695,6 +725,168 @@ def install(config, plugins):
     le_client.enhance_config(domains, config)
 
 
+def _restore_required_config_elements(full_path, config, renewalparams):
+    # string-valued items to add if they're present
+    for config_item in STR_CONFIG_ITEMS:
+        if config_item in renewalparams:
+            value = renewalparams[config_item]
+            # Unfortunately, we've lost type information from ConfigObj,
+            # so we don't know if the original was NoneType or str!
+            if value == "None":
+                value = None
+            setattr(config.namespace, config_item, value)
+    # int-valued items to add if they're present
+    for config_item in INT_CONFIG_ITEMS:
+        if config_item in renewalparams:
+            try:
+                value = int(renewalparams[config_item])
+                setattr(config.namespace, config_item, value)
+            except ValueError:
+                logger.warning("Renewal configuration file %s specifies "
+                               "a non-numeric value for %s. Skipping.",
+                               full_path, config_item)
+                raise
+
+def _restore_plugin_configs(config, renewalparams):
+    # Now use parser to get plugin-prefixed items with correct types
+    # XXX: the current approach of extracting only prefixed items
+    #      related to the actually-used installer and authenticator
+    #      works as long as plugins don't need to read plugin-specific
+    #      variables set by someone else (e.g., assuming Apache
+    #      configurator doesn't need to read webroot_ variables).
+    # XXX: is it true that an item will end up in _parser._actions even
```

Comment:
There is a default; it's `store`: https://docs.python.org/3/library/argparse.html#action
I find `argparse.py` fairly hard to disentangle, so I can't say for sure that this will never be `[]`.  But it looks like Seth's code handles that case okay?  So perhaps we can remove this comment?